ca/ca.go

```diff
@@ -393,29 +412,56 @@ var ocspStatusToCode = map[string]int{
 
 // GenerateOCSP produces a new OCSP response and returns it
 func (ca *CertificateAuthorityImpl) GenerateOCSP(ctx context.Context, req *caPB.GenerateOCSPRequest) (*caPB.OCSPResponse, error) {
-	cert, err := x509.ParseCertificate(req.CertDER)
-	if err != nil {
-		ca.log.AuditErr(err.Error())
-		return nil, err
-	}
-
-	cn := cert.Issuer.CommonName
-	issuer := ca.issuers[cn]
-	if issuer == nil {
-		return nil, fmt.Errorf("This CA doesn't have an issuer cert with CommonName %q", cn)
-	}
+	var issuer *internalIssuer
+	var serial *big.Int
+	// Once the feature is enabled we need to support both RPCs that include
+	// IssuerID and those that don't as we still need to be able to update rows
+	// that didn't have an IssuerID set when they were created. Once this featue
+	// has been enabled for a full OCSP lifetime cycle we can remove this
```

Comment:
I think we actually need to wait for a full certificate lifetime, since issuerID gets set by AddPrecertificate. If we wanted to avoid this waiting period, we could make the migration set a default value for issuerID that matches int-x3.

Reply:
Yeah that's what I mean't, I think I didn't really word it well though.

I prefer doing this hybrid approach rather than setting a default value as it  is somewhat easier to reason about, and we need to have a special migration for our test environment, a special migration for staging, and a special migration for prod, which seems prone to issues.